### ISSUE TITLE
Should the `client-side-routing` example use `aria-live=assertive` for announcing page navigation?

### ISSUE THREAD

**Issue by pete-murphy** [marcysutton/gatsby-a11y-workshop#11]
Hello! We've recently worked on improving the accessibility of navigation within our single-page application, following the advice from [What we learned from user testing of accessible client-side routing techniques with Fable Tech Labs](https://www.gatsbyjs.com/blog/2019-07-11-user-testing-accessible-client-routing):

> - Provide a skip link that takes focus on a route change within the site, with a label that indicates what the link will do when activated: e.g. "skip to main navigation".
> - Include an [ARIA Live Region](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/ARIA_Live_Regions) on page load. On a route change, append text to it indicating the current page, e.g. "Portfolio page".

The article does not mention whether `aria-live=assertive` or `aria-live=polite` should be used for the live region, but it does link to this repo as "an example of the most recent research in action". The code that is linked uses a `<div>` with `role=status` (which has implicit `aria-live=polite`): https://github.com/marcysutton/gatsby-a11y-workshop/blob/master/examples/client-side-routing/gatsby-browser.js#L23-L30

From what I've tested using this approach in our web app, on VoiceOver and NVDA, the focused skip link will be announced and then _sometimes_ the page navigation is announced but sometimes it is omitted. Adding a few hundred milliseconds delay before appending to the live region seems to help with getting the SR to announce the page change more consistently, but looking at the HTML of [testingaccessibility.com](https://testingaccessibility.com/) I see the live region (inside of `<next-route-announcer>`) uses `role=alert` as well as `aria-live=assertive`. Is that the recommended setting?

**Comment by pete-murphy** [comment]
> looking at the HTML of [testingaccessibility.com](https://testingaccessibility.com/) I see the live region (inside of `<next-route-announcer>`) uses `role=alert` as well as `aria-live=assertive`

Hm, testing it out, that site doesn't appear to do any client-side navigation. Oddly, using `assertive` in place of `polite` seems to break our navigation announcer on VoiceOver.

**Comment by pete-murphy** [comment]
Closing this since the `aria-live=polite` seems to work as intended. 

For posterity, I made a minimal CodePen reproducing the issue: https://codepen.io/ptrfrncsmrph-the-vuer/pen/pvodZNR. 

On NVDA, both examples behave the same: the "Clicked {n} times" live region is announced, and then the focused element is announced.

On VoiceOver on Mac, in the `polite` example, the focused element is announced and then the live region is announced.
In the `assertive` example, the live region is not announced at all, just the focused element.
